buildpacks/ruby/src/layers/bundle_install_configure_env_layer.rs

```diff
@@ -0,0 +1,94 @@
+use crate::RubyBuildpackError;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use std::path::Path;
+
+use crate::RubyBuildpack;
+use libcnb::build::BuildContext;
+use libcnb::layer::{Layer, LayerResult, LayerResultBuilder};
+
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+
+/*
+
+# Set up environment for `bundle install`
+
+Must run before `execute_bundle_install_layer`. This is decoupled from dependency
+storage so that it can be updated without having to reinstall gems.
+
+## Layer dir:
+
+None
+
+## Set environment variables
+
+- `BUNDLE_CLEAN=1` - After successful `bundle install` bundler will automatically run `bundle clean`.
+- `BUNDLE_DEPLOYMENT=1` - Requires the `Gemfile.lock` to be in sync with the current `Gemfile`.
+- `BUNDLE_GEMFILE=<app-dir>/Gemfile` - Tells bundler where to find the `Gemfile`.
+- `BUNDLE_WITHOUT=development:test:$BUNDLE_WITHOUT` - Do not install `development` or `test` groups via bundle isntall. Additional groups can be specified via user config.
+
+## Environment NOT set by this buildpack
+
+The following bundler environment variables are set by another layer:
+
+- `BUNDLE_BIN=<layer-dir>/bin` - Install executables for all gems into specified path.
+- `BUNDLE_PATH=<layer-dir>` - Directs bundler to install gems to this path
+- `GEM_PATH=<layer-dir>` - Tells Ruby where gems are located. Should match BUNDLE_PATH.
+
+*/
+pub struct BundleInstallConfigureEnvLayer;
```

Comment:
This layer seems to apply to most or all bundler commands, not just `bundle install`? I feel like this struct name is a bit overly verbose, and you might be able to get away with `BundlerEnvLayer` without losing context?